### ISSUE TITLE
"hmax" returning only nan?

### ISSUE THREAD

**Issue by dumontgoulart** [Deltares/hydromt_sfincs#93]
When generating the 'hmax' raster files in the new hydromt_sfincs version, I noticed they are all returning nan values. This didn't happen in previous versions, and I have not changed the SFINCS version (Alp D'Huez), so I believe it could be something at the new version of hydromt_sfincs. An extra check I did was to plot the 'h' values and these seem to be working okay with actual numerical values.

For reference the sfincs.inp has the attribute "dtmaxout" = 999999, as indicated at the sfincs API (https://sfincs.readthedocs.io/en/latest/parameters.html).

Some code for reference:

```
>>> from os.path import join
>>> import matplotlib.pyplot as plt
>>> import hydromt_sfincs
>>> from hydromt_sfincs import SfincsModel, utils
>>> print(hydromt_sfincs.__version__)
1.0.0
>>> mod = SfincsModel(f'{root_folder}sandy_factual_2_rain_surge', mode="r")
>>> mod.read_results()
>>> mod.write_raster(f"results.hmax", compress="LZW")
Nodata value missing, skipping mask
nodata value missing for D:\paper_3\data\sfincs_ini\spectral_nudging\sandy_factual_2_rain_surge\gis\hmax.tif
>>> # count the number of non-NaN values
>>> num_hmax_non_nans = mod.results['hmax'].count().values
>>> num_h_non_nans = mod.results['h'].count().values
>>> print(f"Number of non-NaN hmax values: {num_hmax_non_nans}")
Number of non-NaN hmax values: 0
>>> print(f"Number of non-NaN h values: {num_h_non_nans}")
Number of non-NaN h values: 15792264
>>> print(mod.results['hmax'].max().values)
nan
>>> print(mod.results['h'].max().values)
114.580986
```

EDIT:

Upon further examination, I think the .inp file is not being correctly imported in the new version, while in the previous version it was (maybe something related to the other issue at https://github.com/Deltares/hydromt_sfincs/pull/88). 

I also ran the code above with the old version, it returns 'hmax' as one 2D raster (no time dimension) and with numerical values as expected :

```
>>> import hydromt_sfincs
/home/hmoreno/.conda/envs/hydromt/lib/python3.10/site-packages/hydromt_sfincs/sfincs.py:8: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:

import os
os.environ['USE_PYGEOS'] = '0'
import geopandas

In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
  import geopandas as gpd
>>> from hydromt_sfincs import SfincsModel, utils
>>>
>>> print(hydromt_sfincs.__version__)
0.2.1
>>> mod = SfincsModel(rf'/gpfs/work3/0/einf4318/sfincs/sandy_shifted_factual_1_rain_surge/', mode="r")
>>> mod.read_results()
>>> mod.results['hmax']
<xarray.DataArray 'hmax' (y: 1355, x: 1538)>
dask.array<where, shape=(1355, 1538), dtype=float32, chunksize=(1355, 1538), chunktype=numpy.ndarray>
Coordinates:
  * x            (x) float32 5.366e+05 5.366e+05 ... 6.902e+05 6.902e+05
  * y            (y) float32 4.445e+06 4.445e+06 4.445e+06 ... 4.58e+06 4.58e+06
    spatial_ref  int64 0
Attributes:
    _FillValue:  -9999.0
>>>
KeyboardInterrupt
>>>
>>> # count the number of non-NaN values
>>> num_hmax_non_nans = mod.results['hmax'].count().values
print(f"Number of non-NaN h values: {num_h_non_nans}")

print(mod.results['hmax'].max().values)
print(mod.results['h'].max().values)>>> num_h_non_nans = mod.results['h'].count().values
>>>
>>> print(f"Number of non-NaN hmax values: {num_hmax_non_nans}")
Number of non-NaN hmax values: 2083990
>>> print(f"Number of non-NaN h values: {num_h_non_nans}")
Number of non-NaN h values: 79348068
>>>
>>> print(mod.results['hmax'].max().values)
80.99306
>>> print(mod.results['h'].max().values)
80.73502
>>>
```
So maybe there is something with the way 'hmax' is being calculated in the new version?

**Comment by roeldegoede** [comment]
Hard to tell what goes wrong, but we have seen some issues lately. Like you mentioned, maybe something with the input goes wrong #88, but I can also imagine that it has to do with this issue #91.

The #88 is already merged into the main, and I expect that we will merge the other open pull requests into the main as well in the coming days. So hopefully that resolves the issue. Anyway, maybe you can share the model with me, then I can have a look at what goes wrong in case it is something different.
